### DISCUSSION TITLE
Is it possible to use a callback when an integer solution is found?

### DISCUSSION THREAD

**Discussion by joelmontoy** [coin-or/Sonnet#22] [Q&A] [answered]
Hello!
I've used CoinMP libraries to solve integer programs before, but now I'm using Sonnet because I love the modelling syntax.
However, sometimes I would like to use callbacks like I could in CoinMP, specially the one that is called when an integer node is found (NodeCallback). 
Is there a way to use CBC callbacks in Sonnet?
Thanks!

**Comment by jhmgoossens** [comment, marked as answer]
Have a look at the [Sonnet_CbcTest5](https://github.com/coin-or/Sonnet/blob/master/test/SonnetTest/Sonnet_CbcTests.cs#L125)  and the eventhandler used there. Does this help?

**Reply by joelmontoy** [reply]
Yes, it definitely helps. Thank you!
However, I get an error saying that enums CbcEvent and CbcAction are not defined in the COIN namespace...
But this is a different problem. I'll try to solve it and I'll open another discussion if I can't! 
![imagen](https://user-images.githubusercontent.com/24981179/148779029-f816063f-8e89-403e-8123-9d0b768c2325.png)

**Reply by jhmgoossens** [reply]
Great! Hint: You need to include `using COIN;` for these, same as done near the top of the Sonnet_CbcTests.cs file.

**Reply by joelmontoy** [reply]
Solved! I had to build the binaries from source. It seems that the precompiled binaries of SonnetWrapper (version 1.4) didn't include CbcEventHandler. Many thanks!
